extensions/amp-action-macro/0.1/amp-action-macro.js

```diff
@@ -14,21 +14,39 @@
  * limitations under the License.
  */
 import {LayoutPriority} from '../../../src/layout';
+import {Services} from '../../../src/services';
 import {isExperimentOn} from '../../../src/experiments';
+import {parseActionMap} from '../../../src/service/action-impl';
+import {toggle} from '../../../src/style';
 import {userAssert} from '../../../src/log';
 
 /** @const {string} */
 const TAG = 'amp-action-macro';
 
 /**
-* The <amp-action-macro> element is used to define a reusable action macro.
+* The <amp-action-macro> element is used to define a reusable action.
 */
 export class AmpActionMacro extends AMP.BaseElement {
 
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?../../../src/service/action-impl.ActionService} */
+    this.actions_ = null;
+  }
+
   /** @override */
   buildCallback() {
     userAssert(isExperimentOn(this.win, 'amp-action-macro'),
         'Experiment is off');
+    toggle(this.element, /* opt_display */ false);
+    this.actions_ = Services.actionServiceForDoc(this.element);
+    this.element.setAttribute('aria-hidden', 'true');
```

Comment:
I guess you got this from amp-state.js, but `toggle()` and `aria-hidden` aren't necessary here. This is because only layout=nodisplay is supported by default -- just remove the `isLayoutSupported` override.